BASE/src/ServerTelemetryChannel/Implementation/TelemetrySerializer.cs

```diff
@@ -27,6 +29,29 @@ public Uri EndpointAddress
         }
 
         public virtual void Serialize(ICollection<ITelemetry> items)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items);
+            this.Transmitter.Enqueue(transmission);
+        }
+
+        public virtual Task<bool> SerializeAsync(ICollection<ITelemetry> items, CancellationToken cancellationToken)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items);
+            Task<bool> resultTask = transmission.GetFlushTask(cancellationToken);
+
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                Task.Run(() => this.Transmitter.Enqueue(transmission), cancellationToken);
+            }
+
+            return resultTask;
```

Comment:
If the cancellationToken was canceled, when is the resultTask canceled?

Reply:
I think it will be cancelled when GetFlushTask is called, as there is the registration for the cancellation in there.

Reply:
correct!